contributors/devel/sig-architecture/alt-arch.md

```diff
@@ -0,0 +1,64 @@
+# Shipping alternate architectures in Kubernetes release artifacts
+
+Kubernetes by default has been amd64 oriented arch-wise. All our build and release systems originally supported just that. A while ago we started an [effort to support multiple architectures](https://github.com/kubernetes/kubernetes/issues/38067). As part of this effort, we added support in our build/release pipelines for arm, arm64, ppc64le and s390x. 
+
+The main focus was to have binaries, container images available for these architectures and for folks that are interested to be able able to take these artifacts and set up CI jobs to adequately test these platforms. Specifically to call out the ability to run conformance tests on these platforms.
+
+So in this document, let's explore with our sig-release/sig-testing hat on, what we are looking for when we talk about a new architecture
+
+# Step 1 : crawling (Build)
+
+- bazel based build infrastructure should support this architecture
+- docker based build infrastructure should support this architecture
+
+The above 2 implicitly means the following
+
+- golang should support the architecture out-of-the-box
+- All our dependencies whether vendored or run separately should support this architecture out-of-the-box
+
+In other words, anyone in the community should be able to use our build infra to generate all artifacts required to stand up Kubernetes.
+
+# Step 2 : walking (Test)
+
+It is not enough for builds to work as it gets bit-rotted quickly when we vendor in new changes, update versions of things we use etc. So we need a good set of tests that exercise a wide battery of tests in this new architecture
+
+- unit tests
+- e2e tests
+- node e2e tests
+
+This will ensure that community members can rely on these architectures on a consistent basis. This will give folks who are making changes a signal when they break things in a specific architecture. 
+
+This implies a set of folks who standup and maintain both post-submit and periodic tests, watch them closely and raise the flag when things break. They will also have to help debug and fix any architecture/plaform specific issues as well. 
+
```

Comment:
Assuming they have raised the flag, what will happen then? Will they fix things, revert things? I suppose this will be on a case by case basis. Should it be the case, I am imagining the possible tensions close to release time whether to revert an important patch because it breaks different architecture, how will that be accepted? How will the architecture team will feel if they are ignored, assuming they never got the change to enter step 3?